src/realm/sort_descriptor.cpp

```diff
@@ -391,6 +402,47 @@ void FilterDescriptor::execute(const Table& table, KeyValues& key_values, const
     key_values = std::move(filtered);
 }
 
+template <typename T, typename U>
+inline int compare(const T& i, const T& j, const U& col)
+{
+    const auto id_i = i.cached_string_id;
+    const auto id_j = j.cached_string_id;
+
+    if (!id_i && !id_j)
+        // any other comparison that has nothing to do with strings
+        return i.get_value().compare(j.get_value());
+
+    if (id_i || id_j) {
+        // at least one is a compressed string.
+        const auto table = col.table;
+        const auto interner = table->get_string_interner(ColKey{col.col_key});
+
+        // two interned strings
+        if (id_i && id_j)
+            return interner->compare(*id_i, *id_j);
```

Comment:
This is going to perform pretty catastrophically badly if multiple threads are trying to sort at once. The extremely fine-grained locking on every comparison could lead to orders-of-magnitude slowdown if the lock is contended.

Reply:
This is a general drawback of how the interner is implemented, and you are right. However, I prefer not to tackle this in this PR, since we have many of this fine locking conditions already.  NOTE: We need to discuss this @finnschiermer

Reply:
Filed for tracking the issue: https://jira.mongodb.org/browse/RCORE-2214